### DISCUSSION TITLE
This package is exactly what I'm looking for!

### DISCUSSION THREAD

**Discussion by ThijSlim** [KonnorRogers/shadow-dom-testing-library#7] [General]
Hi! The previous package `testing-library__dom` stopped working while migrating from Jest to Vitest. Also it is not maintained for the last 2 years. Your package works out of the box when including getByShadowRole.

Something I was missing in the previous package is that when a specific role is not found there was no representation of the DOM like and the available roles inside the (shadow)DOM.

```html
<div>
 <web-component>
 #shadowroot
 <button>Text</button>
</webcomponent>
<div> 
```

The only thing it showed was:

```html
<div>
 <web-component>
 Text
</webcomponent>
<div> 
```

This to improve developer experience. 

Anyways, nice to finally have a package which supports Shadow DOM!

**Comment by KonnorRogers** [comment]
Awesome! Glad to hear that its working well for you! Yea I wanted to be sure to support Vitest so the test suite has the option to run both!

**Comment by cvharris** [comment]
Same. At my job we use [LWC](https://lwc.dev/), a web-component js framework. Previously folks had to chain `querySelector` off `shadowRoot` in order to get the thing they were looking for, so it would never benefit from `testing-library`'s approach to interaction-driven testing. So far, this seems like exactly what we needed for one of our newer projects.

[There is a library](https://github.com/salesforce/kagekiri) one of the LWC devs built that does a lot to help query elements inside a `shadowRoot`, which might have some insights that could benefit this library's purposes. Thank you for publishing this!
